src/entry.c

```diff
@@ -326,6 +326,11 @@ gboolean CacheEntryPasswordSet(CacheEntry *self, const gchar *password,
   return TRUE;
 }
 
+void CacheEntryHashSet(CacheEntry *self, CacheEntryAlgorithm algorithm,
+                  GBytes *hash, CacheEntryArgs *args) {
```

Comment:
I'm not 100% on what you mean here. Isn't g_bytes_ref incrementing the reference count?
https://developer.gnome.org/glib/stable/glib-Byte-Arrays.html#g-bytes-ref

Also, this should be tested in entry_test.c? That is where the reference count should be added?